kartothek/core/dataset.py

```diff
@@ -907,6 +907,37 @@ def from_dataset(dataset):
         ds_builder.partitions = dataset.partitions
         return ds_builder
 
+    def modify_uuid(self, tgt_uuid: str):
+        """
+        Modify the dataset uuid and depending metadata:
+        - paths to partitioning files
+        - path to index files
+
+        :param tgt_uuid:
+            Modified dataset UUID.
+        :return modified builder object
+        """
+
+        # modify file names in partition metadata
```

Comment:
This method is not atomic, i.e. if it fails in the middle of operation it leaves the partitions half modified. That seems a bit fragile to me, perhaps we could build the new partitions and then switch them at the end rather than modifying in place in a loop?

Also I think this method should probably be unit tested.

Reply:
atomicity: OK, changed this. I don't think btw that non-atomicity would be a problem here, since this is a Metadata**Builder** object with a very limited life span, not a long-living Metadata object. But it's not a big change anyway.

unit tests: ok, going to add this. The method is already included in the higher-level unit tests, but dedicated tests for this method make of course sense.